src/services/web3/approval/index.ts

```diff
@@ -89,34 +101,50 @@ const setApproval = async (
 export const getNetworkContractApproval = async (
   token: Token,
   amount: string,
-  contract?: string
+  contract: ApprovalContract | string
 ): Promise<boolean> => {
-  const BANCOR_NETWORK = await bancorNetwork$.pipe(take(1)).toPromise();
-  const USER = await user$.pipe(take(1)).toPromise();
+  const user = await user$.pipe(take(1)).toPromise();
   const amountWei = expandToken(amount, token.decimals);
   const { isApprovalRequired } = await getApproval(
     token.address,
-    USER,
-    contract ? contract : BANCOR_NETWORK,
+    user,
+    await getApprovalAddress(contract),
     amountWei
   );
   return isApprovalRequired;
 };
 
 export const setNetworkContractApproval = async (
   token: Token,
+  contract: ApprovalContract | string,
```

Comment:
I would suggest to keep the order of props the same as `getNetworkContractApproval` to reduce chances of mixing them up, maybe change it to pass an object as prop instead.